src/devices/esp8266.c

```diff
@@ -71,16 +73,58 @@ static void cmd_failure(const char *cmd_name, const char *msg)
 }
 
 /**
+ * Callback that gets invoked when we are unable to handle the URC using
+ * the standard URC callbacks.  Used for the silly messages like
+ * `0,CONNECT` where there is no prefix for the URC like their should be.
+ * Messages this callback handles:
+ * * <0-4>,CONNECT
+ * * <0-4>,CLOSED
+ * * WIFI DISCONNECT
+ */
+static bool sparse_urc_cb(char* msg)
+{
+        msg = strip_inline(msg);
+
+        if (STR_EQ(msg, "WIFI DISCONNECT")) {
+                if (state.hooks.client_wifi_disconnect_cb)
+                        state.hooks.client_wifi_disconnect_cb();
+                return true;
+        }
+
+        /* Now look for a message with a comma */
+        char* comma = strchr(msg, ',');
+        if (!comma)
+                return false;
+
+        *comma = '\0';
+        const char* m1 = msg;
+        const char* m2 = ++comma;
+        if (STR_EQ(m2, "CONNECT")) {
```

Comment:
we can probably safely assume it's a number, and if not, it falls back to zero in the atoi. Thanks.